cpp/src/Ice/Proxy.cpp

```diff
@@ -528,22 +528,14 @@ Ice::ObjectPrx::_setRequestHandler(const ::IceInternal::RequestHandlerPtr& handl
 }
 
 void
-Ice::ObjectPrx::_updateRequestHandler(const ::IceInternal::RequestHandlerPtr& previous,
-                                      const ::IceInternal::RequestHandlerPtr& handler)
+Ice::ObjectPrx::_clearRequestHandler(const ::IceInternal::RequestHandlerPtr& previous)
```

Comment:
Yes, I think so since `_setRequestHandler` only sets `_requestHandler` if the connection is cached. We should actually get rid of `_setRequestHandler` since it's only called by `_getRequestHandler`.